lib/scope.js

```diff
@@ -562,7 +562,16 @@ module.exports = {
         set_to: row.value,
       });
       if ( handle ) {
-        fields_to_set.push({ row, handle, tag, type });
+        if ( (tag === 'CANVAS') && row.value === '/sign' ) {
+          // A signature field on a signature page is the only field on that page
+          fields_to_set.push({ row, handle, tag, type });
+          // Get the rest of the variables before...
+          remaining_vars = remaining_vars.concat( var_data.slice( var_data.indexOf(row) + 1 ));
```

Comment:
Is it a safe assumption to make that we always just want the first `/sign` in the var_data array? It seems safe enough, but wanted to flag it in case this raises any concerns for you.

Reply:
All `/sign` rows are the same currently, so it matters even less than usual (since tables are supposed to not care about order).